docs: revamp README with comprehensive example usage and guidance

appleboy opened the pull request:
- Rewrite and expand the README to provide a detailed overview of the example categories, launch options, and usage instructions
- Add an extensive table of contents with clear links to all major example types
- Document step-by-step instructions for running examples in various modes (console, restapi, a2a, webui)
- Clarify environment variable requirements and configuration patterns across examples
- Distinguish between full and production launchers, specifying their supported modes
- Add sections on contributing, common patterns, and integration prerequisites for advanced examples
- Improve consistency, clarity, and completeness throughout the guide

appleboy committed:
docs: revamp README with comprehensive example usage and guidance

- Rewrite and expand the README to provide a detailed overview of the example categories, launch options, and usage instructions
- Add an extensive table of contents with clear links to all major example types
- Document step-by-step instructions for running examples in various modes (console, restapi, a2a, webui)
- Clarify environment variable requirements and configuration patterns across examples
- Distinguish between full and production launchers, specifying their supported modes
- Add sections on contributing, common patterns, and integration prerequisites for advanced examples
- Improve consistency, clarity, and completeness throughout the guide

Signed-off-by: appleboy <appleboy.tw@gmail.com>

kdroste-google commented on examples/README.md:
```diff
@@ -1,27 +1,254 @@
+go run ./examples/<example-name>/main.go a2a
+
+# Run with Web UI
+go run ./examples/<example-name>/main.go webui
```
webui => web api webui

kdroste-google commented on examples/README.md:
```diff
@@ -1,27 +1,254 @@
+go run ./examples/a2a/main.go a2a
+
+# In another terminal, connect as client
+go run ./examples/a2a/main.go a2a --client
```
Won't work. But something similar might be useful.

kdroste-google commented on examples/README.md:
````diff
@@ -1,27 +1,254 @@
+
+```bash
+# Start the A2A server
+go run ./examples/a2a/main.go a2a
````
a2a => web a2a

kdroste-google commented on examples/README.md:
```diff
@@ -1,27 +1,254 @@
+
+- `GOOGLE_API_KEY`: Gemini API key for model access
+- `GOOGLE_CLOUD_PROJECT`: GCP project ID (for Vertex AI)
+- `PORT`: Server port (default: 8080)
```
Not really, You may specify port for web launcher.

kdroste-google reviewed (changes requested):
Please review the file, the content at the moment is not 100% accurate for ADK Go.

appleboy committed:
docs: correct examples README per review feedback

Address reviewer comments (@kdroste-google, gemini-code-assist) on the
examples README so the documented commands and snippets actually compile
and run:

- Fix launch modes: restapi/a2a/webui -> web api / web a2a / web api webui;
  explain that the web launcher hosts api/a2a/webui sub-servers and the
  Web UI is served by the api server
- a2a example: it starts an in-process A2A server and connects as a remote
  agent; drop the non-existent server/client --client flow
- Note any example can be launched with `web api webui`
- sequentialCode: describe it as a sequential code-writing workflow
  (writer -> reviewer -> refactorer)
- Launcher snippets: ParseAndRun -> Execute, FormatSyntax -> CommandLineSyntax
- Config snippet: AgentCreator -> AgentLoader (agent.NewSingleLoader)
- Port is set via the web launcher `-port` flag, not a PORT env var

wolo-lab reviewed (changes requested):
I went through the new README line by line against current `main` (`70e2f03c1f2c221018c0d6a22ee8e48559dd7523`) and ran every command it documents. It is a clear improvement on what is in the tree today — the current file still documents `l.ParseAndRun` and `l.FormatSyntax`, and `FormatSyntax` no longer exists on any type — so I would like to land this. I found 14 problems, five of which are commands or behavior that do not work as written.

**The branch is 97 commits behind `main`.** The base here is from 10 June. `examples/README.md` itself did not change on `main`, which is why GitHub still shows it as mergeable, but the `examples/` tree it describes did. A rebase is the first step, because most of what follows falls out of that gap.

**The README documents 11 of about 40 example directories.** With a table of contents and an "Advanced Examples" heading, the omissions read as "these do not exist". Missing are the whole `examples/workflow/` graph engine (10 runnable examples, and it has [its own README](https://github.com/google/adk-go/blob/70e2f03c1f2c221018c0d6a22ee8e48559dd7523/examples/workflow/README.md)), plus `bidi`, `multiagent`, `agentregistry`, `agentengine`, `openai`, `rest`, `skills`, `telemetry`, `toolconfirmation`, `tools/loadmemory`, and `vertexai/vertexengine`. Either cover them or drop the framing that this is a complete index — your call, but the current shape promises more than it delivers.

**[`web api -port 9090`](https://github.com/google/adk-go/blob/fdfd48628ee57065f75cbbde2f157682db855d26/examples/README.md#L233) fails.** `-port` belongs to the `web` launcher's flagset, [defined here](https://github.com/google/adk-go/blob/70e2f03c1f2c221018c0d6a22ee8e48559dd7523/cmd/launcher/web/web.go#L251), not to `api`'s. Go's `flag` stops at the first non-flag argument, so `-port 9090` reaches the `api` sublauncher, which does not define it. The working form is `web -port 9090 api`.

```
$ go run ./examples/quickstart/ web api -port 9090
flag provided but not defined: -port
...
2026/08/25 14:21:33 Run failed: the "api" launcher cannot parse arguments: failed to parse api flags: flag provided but not defined: -port
```

**["The Web UI is served by the `api` server"](https://github.com/google/adk-go/blob/fdfd48628ee57065f75cbbde2f157682db855d26/examples/README.md#L38) is not how it works**, and the same claim is repeated [in the launcher section](https://github.com/google/adk-go/blob/fdfd48628ee57065f75cbbde2f157682db855d26/examples/README.md#L197). `webui` is a sibling sublauncher that registers `/ui/` on the shared router, [see webui.go](https://github.com/google/adk-go/blob/70e2f03c1f2c221018c0d6a22ee8e48559dd7523/cmd/launcher/web/webui/webui.go#L117-L122). Running `web webui` with no `api` serves the UI just fine — `/ui/` returns 200 and `/` redirects to it — but `/api/list-apps` is a 404. So `api` is needed because the browser app calls the REST backend, not to bring the UI up.

**The `go run ./examples/<name>/main.go` form should be `go run ./examples/<name>/`.** Every other README in the repo uses the directory form, and the file form breaks for any example whose `main` package spans more than one file:

```
$ go run ./examples/workflow/complex/main.go console
# command-line-arguments
examples/workflow/complex/main.go:161:17: undefined: newResilientModel
examples/workflow/complex/main.go:188:16: undefined: newResilientModel
```

[`help`](https://github.com/google/adk-go/blob/fdfd48628ee57065f75cbbde2f157682db855d26/examples/README.md#L35) is not a command. It matches no sublauncher keyword, so the run fails with `cannot parse following arguments: [help]` and exits 1. The usage text does get printed, but only on the fatal path, so documenting it as the way to get help is misleading. This one is inherited from the current README rather than introduced here.

Smaller things, all verified against `main`:

- The quickstart section [lists "code execution, search"](https://github.com/google/adk-go/blob/fdfd48628ee57065f75cbbde2f157682db855d26/examples/README.md#L52), but [quickstart registers only `geminitool.GoogleSearch`](https://github.com/google/adk-go/blob/70e2f03c1f2c221018c0d6a22ee8e48559dd7523/examples/quickstart/main.go#L49-L51).
- [`tools/multipletools` is described as registering multiple custom tools](https://github.com/google/adk-go/blob/fdfd48628ee57065f75cbbde2f157682db855d26/examples/README.md#L116). It is really a workaround for combining Google Search with a custom tool, which is why it [wraps two sub-agents in `agenttool.New`](https://github.com/google/adk-go/blob/70e2f03c1f2c221018c0d6a22ee8e48559dd7523/examples/tools/multipletools/main.go#L99-L104).
- The [environment variable list](https://github.com/google/adk-go/blob/fdfd48628ee57065f75cbbde2f157682db855d26/examples/README.md#L226-L227) is missing `GOOGLE_CLOUD_LOCATION` and `VERTEX_ENGINE_ID`, both fatal when unset in the `vertexai` examples, and `GITHUB_PAT` / `AGENT_MODE` for `mcp`. Naming `GITHUB_PAT` would also make the `mcp` prerequisite concrete.
- [The docs link](https://github.com/google/adk-go/blob/fdfd48628ee57065f75cbbde2f157682db855d26/examples/README.md#L267) points at `pkg.go.dev/google.golang.org/adk`. The module is `google.golang.org/adk/v2`.
- `tools/loadartifacts` [has to be run from its own directory](https://github.com/google/adk-go/blob/70e2f03c1f2c221018c0d6a22ee8e48559dd7523/examples/tools/loadartifacts/main.go#L37-L38) because it reads its image by relative path, so the run-from-repo-root instruction does not hold there.
- The [`vertexai` subdirectory list](https://github.com/google/adk-go/blob/fdfd48628ee57065f75cbbde2f157682db855d26/examples/README.md#L170-L173) names only `imagegenerator`. There is also `vertexengine/` and a root `agent.go`.
- [`if err = l.Execute(...)`](https://github.com/google/adk-go/blob/fdfd48628ee57065f75cbbde2f157682db855d26/examples/README.md#L187) uses an undeclared `err`. The [equivalent snippet further down](https://github.com/google/adk-go/blob/fdfd48628ee57065f75cbbde2f157682db855d26/examples/README.md#L250) uses `:=`.
- The table of contents omits Common Patterns, Contributing, and Additional Resources.

Everything else checked out: `full.NewLauncher`, `prod.NewLauncher`, `launcher.Config.AgentLoader`, `agent.NewSingleLoader`, `l.Execute`, `l.CommandLineSyntax`, the `prod` mode list, the `web <sub>` grammar, and the `sequentialCode` writer to reviewer to refactorer description are all accurate, and `console`, `web api webui`, and `web a2a` all run as documented.
